src/operations/identify.rs

```diff
@@ -2,41 +2,88 @@ use std::{ffi::OsStr, io::Write};
 
 use image::ExtendedColorType;
 
-use crate::{error::MagickError, image::Image, wm_try};
+use crate::{
+    arg_parsers::IdentifyFormat, arg_parsers::Token, arg_parsers::Var, error::MagickError,
+    image::Image, wm_try,
+};
 
 // https://imagemagick.org/script/command-line-options.php#identify
-pub fn identify(image: &mut Image) -> Result<(), MagickError> {
+pub fn identify(image: &mut Image, format: IdentifyFormat) -> Result<(), MagickError> {
     // acquire a buffered writer to which we can make lots of small writes cheaply
     let mut stdout = std::io::stdout().lock();
-    identify_impl(image, &mut stdout)
+    identify_impl(image, format, &mut stdout)
 }
 
-fn identify_impl(image: &Image, writer: &mut impl Write) -> Result<(), MagickError> {
-    write_filename(&image.properties.filename, writer)?;
+fn identify_impl(
+    image: &Image,
+    format: IdentifyFormat,
+    writer: &mut impl Write,
+) -> Result<(), MagickError> {
+    // The default format, if none is specified, turns into something like
+    // ~/imagename.jpg JPEG 1363x2048 1363x2048+0+0 8-bit sRGB 270336B 0.010u 0:00.013
+    let template = &format.template.unwrap_or(vec![
```

Comment:
I could've also decided to not use an optional for the template and instead put the default vector into the structure from the beginning (using the `Default` trait?). No strong feelings to be honest.